app/src/main/java/com/story/mipsa/attendancetracker/MainActivity.java

```diff
@@ -369,28 +327,111 @@ public void OnItemClick(int position) {
-
-                    public void onClick(DialogInterface arg0, int arg1) {
-                        MainActivity.super.onBackPressed();
-                    }
-                }).create().show();
+        if(isTaskRoot()){
+            new AlertDialog.Builder(this)
+                    .setTitle("Exit?")
+                    .setMessage("Are you sure you want to exit?")
+                    .setNegativeButton(android.R.string.no, null)
+                    .setPositiveButton(android.R.string.yes, new DialogInterface.OnClickListener() {
+                        public void onClick(DialogInterface arg0, int arg1) {
+                            MainActivity.super.onBackPressed();
+                        }
+                    }).create().show();
+        }
+        else{
+            super.onBackPressed();
+            overridePendingTransition(R.anim.fade_in, R.anim.fade_out);
+        }
     }
 
 
 
     @Override
     public void onDateSelected(DateTime dateSelected) {
-//        Toast.makeText(getApplicationContext(), dateSelected.toString(), Toast.LENGTH_SHORT).show();
-        SimpleDateFormat sdf = new SimpleDateFormat("d MMM yyyy, EEE");
-//        selectedDate = sdf.format(dateSelected.toDate());
         setSelectedDate(dateSelected.getMillis());
-//        Toast.makeText(getApplicationContext(),""+dateSelected.getMillis(), Toast.LENGTH_SHORT).show();
+    }
+
+    @Override
+    public void sendExtraInput(String status, int position) {
+        insertExtraClass(status, position);
+    }
+
+    private void insertExtraClass(String status, int position) {
+        SubjectItem current = subjectItems.get(position);
+        if (status.equalsIgnoreCase("Present")) {
+            current.setPresent(current.getPresent() + 1);
+            current.setTotal(current.getTotal() + 1);
+            current.setSubjectAttendanceDetails(new SubjectAttendanceDetails(status, getSelectedDate(), true));
+            Recalculate(current);
+            user = firebaseAuth.getCurrentUser();
+            DatabaseReference userRef = ref.child("Users");
+            userRef.child(user.getUid()).child("Subjects").child(current.getSubjectName()).setValue(current);
+            buildRecyclerView();
+
+        } else if (status.equalsIgnoreCase("Absent")) {
+            current.setAbsent(current.getAbsent() + 1);
+            current.setTotal(current.getTotal() + 1);
+            current.setSubjectAttendanceDetails(new SubjectAttendanceDetails(status, getSelectedDate(), true));
+            Recalculate(current);
+            user = firebaseAuth.getCurrentUser();
+            DatabaseReference userRef = ref.child("Users");
+            userRef.child(user.getUid()).child("Subjects").child(current.getSubjectName()).setValue(current);
+            buildRecyclerView();
+        }
+    }
+
+    private void Recalculate(SubjectItem current) {
+        int presentS = current.getPresent();
+        int total = current.getTotal();
+        int totalS = total;
```

Comment:
Are two variables required? Can't you do `int totalS = current.getTotal();` directly?